core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -423,11 +424,19 @@ protected void setup(Binder binder)
         // split monitor
         binder.bind(SplitMonitor.class).in(Scopes.SINGLETON);
 
+        // This replicates logic for computing general pool size from LocalMemoryManager
+        // TODO: figure out if it is possible to use Guice to bind common value in LocalMemoryManager and to discoveryBinder
```

Comment:
Yeah - but it would make `configureMemoryPools` in `LocalMemoryManager` less readable imo (the lines to be extracted are overlapped now with configuring reserved pool): https://github.com/trinodb/trino/blob/9ab09e99d6c56fe955697a51a95400c367525133/core/trino-main/src/main/java/io/trino/memory/LocalMemoryManager.java#L57-L70
I can if you insist.